br/pkg/lightning/backend/local/localhelper.go

```diff
@@ -397,31 +398,46 @@ func (local *local) BatchSplitRegions(
 	var failedErr error
 	retryRegions := make([]*split.RegionInfo, 0)
 	scatterRegions := newRegions
-	waitTime := splitRegionBaseBackOffTime
-	for i := 0; i < maxRetryTimes; i++ {
+	backoffer := split.NewWaitRegionOnlineBackofferWithVars(
+		30, time.Second, 10*time.Second).(*split.WaitRegionOnlineBackoffer)
+	_ = utils.WithRetry(ctx, func() error {
+		retryRegions = make([]*split.RegionInfo, 0, 16)
 		for _, region := range scatterRegions {
 			// Wait for a while until the regions successfully splits.
-			local.waitForSplit(ctx, region.Region.Id)
+			ok, err2 := local.hasRegion(ctx, region.Region.Id)
+			if !ok || err2 != nil {
+				failedErr = err2
+				if failedErr == nil {
+					failedErr = errors.Errorf("region %d not exists", region.Region.Id)
+				}
+				retryRegions = append(retryRegions, region)
+				continue
+			}
 			if err = local.splitCli.ScatterRegion(ctx, region); err != nil {
 				failedErr = err
 				retryRegions = append(retryRegions, region)
 			}
 		}
 		if len(retryRegions) == 0 {
-			break
+			return nil
+		}
+		// if the number of becomes smaller, we can infer TiKV side really
+		// made some progress so don't increase the retry times.
+		if len(retryRegions) < len(scatterRegions) {
+			backoffer.Stat.ReduceRetry()
 		}
 		// the scatter operation likely fails because region replicate not finish yet
 		// pack them to one log to avoid printing a lot warn logs.
 		log.FromContext(ctx).Warn("scatter region failed", zap.Int("regionCount", len(newRegions)),
-			zap.Int("failedCount", len(retryRegions)), zap.Error(failedErr), zap.Int("retry", i))
+			zap.Int("failedCount", len(retryRegions)), zap.Error(failedErr))
 		scatterRegions = retryRegions
-		retryRegions = make([]*split.RegionInfo, 0)
-		select {
-		case <-time.After(waitTime):
-		case <-ctx.Done():
-			return nil, nil, ctx.Err()
-		}
-		waitTime *= 2
+		return failedErr
+	}, backoffer)
+
+	select {
+	case <-ctx.Done():
+		return nil, nil, ctx.Err()
+	default:
 	}
```

Comment:
```suggestion
```

Reply:
do you mean we can `return region, newRegions, ctx.Err()`?

Reply:
return nil is enough.

Reply:
There is no blocking call here, it's no need to check the context.

Reply:
I noticed that caller have checked context error `IsContextCanceledError`, maybe it's related to some network blocking? And if we return nil result and nil error to caller it may behave abnormally. So I prefer we return `return region, newRegions, ctx.Err()`